astropy/tests/helper.py

```diff
@@ -304,14 +304,27 @@ def _save_coverage(cov, result, rootdir, testing_path):
     # Python 2.x.
     if six.PY2:
         d = cov.data
-        cov._harvest_data()
-        for key in d.lines.keys():
-            new_path = os.path.relpath(
-                os.path.realpath(key),
-                os.path.realpath(testing_path))
-            new_path = os.path.abspath(
-                os.path.join(rootdir, new_path))
-            d.lines[new_path] = d.lines.pop(key)
+        try:
+            # Coverage 4.0: _harvest_data has been renamed to get_data, the
+            # lines dict is private
+            cov.get_data()
+            for key in d._lines.keys():
+                new_path = os.path.relpath(
+                    os.path.realpath(key),
+                    os.path.realpath(testing_path))
+                new_path = os.path.abspath(
+                    os.path.join(rootdir, new_path))
+                d._lines[new_path] = d._lines.pop(key)
+        except AttributeError:
+            # Coverage < 4.0
+            cov._harvest_data()
+            for key in d.lines.keys():
+                new_path = os.path.relpath(
+                    os.path.realpath(key),
+                    os.path.realpath(testing_path))
+                new_path = os.path.abspath(
+                    os.path.join(rootdir, new_path))
+                d.lines[new_path] = d.lines.pop(key)
```

Comment:
Thanks for trying to fix this.  This duplicates lots of code though, and there's lots of places where *maybe* an `AttributeError` could occur, so it's not clear with this try/except structure that it's cleanly doing different things for two different versions of the coverage library.

Instead I'd do something like:
```python
try:
    cov.get_data()
except AttributeError:
    cov._harvest_data()
    lines = d.lines
else:
    lines = d._lines

# Do stuff with 'lines'
```